### ISSUE TITLE
Menubar tabindex and aria semantics have to be refined

### ISSUE THREAD

**Issue by jooyoungseo** [rstudio/rstudio#6858]
### System details

    RStudio Edition : Server
    RStudio Version : 1.3.957
    OS Version      : Windows and NVDA in Chrome
    R Version       : any

### Steps to reproduce the problem

1. Load NVDA.
2. Open RStudio Server and make sure the following options are enabled in `Help / Accessibility / Accessibility Options...`:

  - [x] Screen reader support (requires restart)
  - [x] Entire page has application role (requires restart)

3. Press CTRL+2 to focus in console.
4. Press Alt+Shift+F to focus on "File" menu.
5. Hit down-arrow key several times to see if NVDA focuses on and reads correct items.

### Describe the problem in detail

NVDA does not focus on and read menu items correctly.

This issue has been caused by the fact that NVDA is not forced to switch on "Focus Mode" for the menu bar and it remains in "Browse Mode." In other words, if you explicitly turn on "Focus Mode" by pressing NVDA+Spacebar before accessing the menu bar, this issue will be resolved.

FYI, JAWS 2020 does not have this issue currently.

### Describe the behavior you expected

It seems like we need to clean up the code for menu bar semantics to better support NVDA.

I have thus far found [Navigation Menubar Example](https://w3c.github.io/aria-practices/examples/menubar/menubar-editor.html) most satisfactory for both NVDA and JAWS.

If you compare RStudio Server's source code to the example above, you would find that we need to explicitly assign `tabindex = "-1"` attribute for all the elements having `role = "menuitem"`.

The only exception, however, is that we need `tabindex = "0"` for the very first `menuitem` element (in our case, `File` menu).

Currently, I see `tabindex = "0"` assigned to the parent container having `role = "menubar"` attribute.

Two requests I have are as follows:

1. Would you please fix the `tabindex` attribute of the menu bar following the example? (removing `tabindex` from the parent `menubar` container and assigning "-1" value for each of the `menuitem" elements while giving "0" value for the very first one.)
2. Could you please let me know whether our menu bar has the `role = "menu"` aria semantic defined for each menu (e.g., File, Edit, ...), which contains `role = "menuitem"`? I am asking this because I was not able to find `role = "menu"` attribute (not `menuitem`) for each menu.

You may understand what I am talking about from the example I have shared above.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
When I tried, NVDA seemed to be working alright now (maybe due to an NVDA and/or Chrome update since this was opened). However, looking at the menu implemention reveals we do deviate significantly from the recommended pattern described here:

https://w3c.github.io/aria-practices/#menu

## Summary of current implementation

The menubar (a GWT component) is implemented using the aria-activedescendant model where the menubar actually keeps keyboard focus but updates aria-activedescendant to indicate which menu item is selected as you use arrow keys.

The top level (the menubar containing File, Edit, Code, etc.) is a div with tabindex=0, role=menubar, and aria-activedescendant.

It contains items with role=menuitem and aria-haspopup=true. These are the "File", "Edit", Code" menu items themselves.

When you open the File menu via keyboard shortcut, focus is placed on the top-level menubar div, and aria-activedescendant refers to the element ID of the File menuitem, and the File menuitem gains an aria-expanded=true attribute.

At the same time, a new div is dynamically created (near the bottom of the HTML just before the closing body tag) with role=menu, tabindex=-1, and its own aria-activedescendant. This is not a child element of the "File" menu (violated recommended pattern, see below). Initially we don't give it focus when the menu was opened via keyboard shortcut (we are supposed to, see below).

Inside this role=menu div are the items with role="menuitem". Some of these have their own submenus so also have aria-haspopup-true.

##  Variations from recommended pattern

- when we open a menu via keyboard shortcut (e.g. open the File menu) we don't put focus on the first menu item in the popup; instead it remains on the "File" menu itself and you must hit the down arrow to move focus to the menu
- when a menuitem opens a submenu (File opens file menu, File / New File opens a submenu, etc.), that submenu is supposed to be contained in the same menu element as the menuitem that opened it, and be immediate sibling of the menuitem that opened it; neither is true; so we might need to dynamically set aria-owns here to establish this relationship
- hitting escape when a submenu is displayed collapses all the popups, not just the current one (tracked here: #6327)
